src/service/rest/result.rs

```diff
@@ -0,0 +1,80 @@
+use diesel;
+use diesel::result::{Error as DieselError, QueryResult};
+use rocket::http::Status;
+use rocket::request::Request;
+use rocket::response::{Responder, Response};
+use rocket_contrib::Json;
+use serde::{Serialize, Serializer};
+use serde::ser::SerializeMap;
+use serde_json;
+use std::convert::From;
+use std::io::Cursor;
+
+#[derive(Debug)]
+pub struct ApiError {
+    http_status: u16,
```

Comment:
I think it's perfectly fine to use `u16` here. But when you eventually move to constants you should know that rocket provides them. https://api.rocket.rs/rocket/http/struct.Status.html

Reply:
Maybe I should just put the `Status` in here. The reason I have this is so that the service layer can return status codes along with error messages. Any decoupling gained by using `u16` here is probably already undermined by the fact that the service layer would be choosing the response code for the `ApiResult`